distributed/s3.py

```diff
@@ -65,6 +67,66 @@ def get_list_of_summary_objects(bucket_name, prefix='', delimiter='',
     return [s for s in L if s.key[-1] != '/']
 
 
+def seek_delimiter(ob, offset, delimiter, seeklength=4096):
+    """
+    Find the bytes location of the next delimiter after the given offset
+
+    Parameters
+    ----------
+    ob : boto3 open s3 key
+    offset : int
+        bytes location to start
+    delimiter : bytes
+        pattern to seek
+    seeklength : int
+        If seeking delimited, the blocksize for finding delimiters.
+    """
+    if offset == 0:
+        return 0
+    data = b''
+    while True:
+        try:
+            piece = ob.get(Range="bytes=%i-%i" %
+                           (offset, offset + seeklength))['Body'].read()
+        except ClientError:
+            return ob.get()['ContentLength']
+        data += piece
+        try:
+            i = data.index(delimiter)
+            return offset + i + len(delimiter)
+        except ValueError:
+            offset += seeklength
```

Comment:
Does this handle all of the cases that `hdfs3.utils.seek_delimiter` handles?  I sometimes get a bit nervous whenever I see two copies of difficult logic.